pkg/controller/job/job_controller.go

```diff
@@ -513,9 +548,46 @@ func (jm *JobController) syncJob(key string) error {
 			return err
 		}
 	}
+
+	if jobHaveNewFailure {
+		jm.backoff.Next(key, time.Duration(*job.Spec.BackoffSeconds), time.Now())
+		// re-enqueue Job after the backoff period
+		jm.enqueueJob(&job)
+	} else {
+		// if no new Failure the job backoff period can be reset
+		jm.backoff.Reset(key)
+	}
+
 	return manageJobErr
 }
 
+func jobToManyFailed(nbRetry int32, maxRetry *int32) bool {
```

Comment:
See my previous comment, this method is not needed.

Reply:
method removed